src/models/connectionStore.ts

```diff
@@ -137,29 +137,26 @@ export class ConnectionStore {
 		return this.loadProfiles(getWorkspaceProfiles);
 	}
 
-	public addSavedPassword(credentialsItem: IConnectionCredentialsQuickPickItem): Promise<IConnectionCredentialsQuickPickItem> {
+	public async addSavedPassword(credentialsItem: IConnectionCredentialsQuickPickItem): Promise<IConnectionCredentialsQuickPickItem> {
 		let self = this;
-		return new Promise<IConnectionCredentialsQuickPickItem>((resolve, reject) => {
-			if (typeof (credentialsItem.connectionCreds['savePassword']) === 'undefined' ||
-				credentialsItem.connectionCreds['savePassword'] === false) {
-				// Don't try to lookup a saved password if savePassword is set to false for the credential
-				resolve(credentialsItem);
-				// Note that 'emptyPasswordInput' property is only present for connection profiles
-			} else if (self.shouldLookupSavedPassword(<IConnectionProfile>credentialsItem.connectionCreds)) {
-				let credentialId = ConnectionStore.formatCredentialIdForCred(credentialsItem.connectionCreds, credentialsItem.quickPickItemType);
-				self._credentialStore.readCredential(credentialId)
-					.then(savedCred => {
-						if (savedCred) {
-							credentialsItem.connectionCreds.password = savedCred.password;
-						}
-						resolve(credentialsItem);
-					})
-					.catch(err => reject(err));
+		if (typeof (credentialsItem.connectionCreds['savePassword']) === 'undefined' ||
+			credentialsItem.connectionCreds['savePassword'] === false) {
+			// Don't try to lookup a saved password if savePassword is set to false for the credential
+			return credentialsItem;
+			// Note that 'emptyPasswordInput' property is only present for connection profiles
+		} else if (self.shouldLookupSavedPassword(<IConnectionProfile>credentialsItem.connectionCreds)) {
+			let credentialId = ConnectionStore.formatCredentialIdForCred(credentialsItem.connectionCreds, credentialsItem.quickPickItemType);
+			const savedCred = await self._credentialStore.readCredential(credentialId);
+			if (savedCred) {
+				credentialsItem.connectionCreds.password = savedCred.password;
+				return credentialsItem;
 			} else {
-				// Already have a password, no need to look up
-				resolve(credentialsItem);
+				Promise.reject(new Error('No saved password found'));
```

Comment:
This isn't doing anything - Promise.reject just returns a promise that's rejected with the given value. But it's up to you to do something with that promise. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject

```suggestion
				throw new Error('No saved password found');
```